This patch addresses an issue with the snmpd daemon crashing
when a proxy subagent host is unavailable.
Upstream fix details at
https://sourceforge.net/p/net-snmp/code/ci/6fd7499ccaafdf244a74306972562b2091cb91b1/

--- net-snmp-5.8.old/agent/mibgroup/ucd-snmp/proxy.c	2018-07-16 07:33:40.000000000 +0000
+++ net-snmp-5.8/agent/mibgroup/ucd-snmp/proxy.c	2021-04-20 04:18:21.980126080 +0000
@@ -569,6 +569,17 @@
     }
 
     switch (operation) {
+    case NETSNMP_CALLBACK_OP_RESEND:
+         /*
+         * Issue#147: Net-SNMP not responding when proxy requests times out
+         *
+         * When snmp_api issue a resend, the default case was hit and the 
+         * delagated cache was freed.
+         * As a result, the NETSNMP_CALLBACK_OP_TIMED_OUT never came in.
+         */
+        DEBUGMSGTL(("proxy", "pdu has been resent for request = %8p\n", requests));
+        return SNMP_ERR_NOERROR;
+
     case NETSNMP_CALLBACK_OP_TIMED_OUT:
         /*
          * WWWXXX: don't leave requests delayed if operation is
